File: rgctl-tests/ecommerce-go/internal/models/order.go
package models

type Order struct {
	ID         string `gorm:"primaryKey" json:"id"`
	UserID     string `gorm:"not null;index" json:"user_id"`
	Status     string `gorm:"not null" json:"status"`
	TotalCents int64  `gorm:"not null" json:"total_cents"`
	CreatedAt  string `gorm:"not null" json:"created_at"`
}

func (o *Order) SetCreatedAt(v string) { o.CreatedAt = v }

type OrderItem struct {
	ID              string `gorm:"primaryKey" json:"id"`
	OrderID         string `gorm:"not null;index" json:"order_id"`
	ProductID       string `gorm:"not null" json:"product_id"`
	Quantity        int64  `gorm:"not null" json:"quantity"`
	UnitPriceCents  int64  `gorm:"not null" json:"unit_price_cents"`
}

File: rgctl-tests/ecommerce-python/app/models/order.py
from sqlalchemy import ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base
from app.models.base import TimestampMixin


class Order(Base, TimestampMixin):
    __tablename__ = "orders"

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    user_id: Mapped[str] = mapped_column(String(36), ForeignKey("users.id"), index=True)
    status: Mapped[str] = mapped_column(String(50))
    total_cents: Mapped[int] = mapped_column(Integer)

    user = relationship("User", back_populates="orders")
    items = relationship("OrderItem", back_populates="order", cascade="all, delete-orphan")


class OrderItem(Base):
    __tablename__ = "order_items"

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    order_id: Mapped[str] = mapped_column(String(36), ForeignKey("orders.id"), index=True)
    product_id: Mapped[str] = mapped_column(String(36), ForeignKey("products.id"))
    quantity: Mapped[int] = mapped_column(Integer)
    unit_price_cents: Mapped[int] = mapped_column(Integer)

    order = relationship("Order", back_populates="items")
    product = relationship("Product", back_populates="order_items")
